x/vesting/types/vesting_account_test.go

```diff
@@ -0,0 +1,848 @@
+	require.Nil(t, plva.DelegatedFree)
+	require.Equal(t, sdk.Coins{sdk.NewInt64Coin(stakeDenom, 25)}, plva.DelegatedVesting)
+}
+
+func TestGetVestedCoinsCliffVestingAcc(t *testing.T) {
+	now := tmtime.Now()
+	cliffTime := now.Add(12 * time.Hour)
+	endTime := now.Add(24 * time.Hour)
+
+	bacc, origCoins := initBaseAccount()
+	cva := types.NewCliffVestingAccount(bacc, origCoins, now.Unix(), cliffTime.Unix(), endTime.Unix())
+
+	// require no coins vested in the very beginning of the vesting schedule
+	vestedCoins := cva.GetVestedCoins(now)
+	require.Nil(t, vestedCoins)
+
+	// require no coins vested in the very beginning of the cliff schedule
+	vestedCoins = cva.GetVestedCoins(cliffTime)
+	require.Nil(t, vestedCoins)
+
+	// require all coins vested at the end of the vesting schedule
+	vestedCoins = cva.GetVestedCoins(endTime)
+	require.Equal(t, origCoins, vestedCoins)
+
+	// require 75% of coins vested
+	vestedCoins = cva.GetVestedCoins(now.Add(18 * time.Hour))
+	require.Equal(t, sdk.Coins{sdk.NewInt64Coin(feeDenom, 750), sdk.NewInt64Coin(stakeDenom, 75)}, vestedCoins)
+
+	// require 100% of coins vested
+	vestedCoins = cva.GetVestedCoins(now.Add(48 * time.Hour))
+	require.Equal(t, origCoins, vestedCoins)
+}
+
+func TestGetVestingCoinsCliffVestingAcc(t *testing.T) {
+	now := tmtime.Now()
+	cliffTime := now.Add(12 * time.Hour)
+	endTime := now.Add(24 * time.Hour)
+
+	bacc, origCoins := initBaseAccount()
+	cva := types.NewCliffVestingAccount(bacc, origCoins, now.Unix(), cliffTime.Unix(), endTime.Unix())
+
+	// require all coins vesting in the beginning of the vesting schedule
+	vestingCoins := cva.GetVestingCoins(now)
+	require.Equal(t, origCoins, vestingCoins)
+
+	// require all coins vesting in the very beginning of the cliff vesting schedule
+	vestingCoins = cva.GetVestingCoins(cliffTime)
+	require.Equal(t, origCoins, vestingCoins)
+
+	// require no coins vesting at the end of the vesting schedule
+	vestingCoins = cva.GetVestingCoins(endTime)
+	require.Nil(t, vestingCoins)
+
+	// require 25% of coins vesting
+	vestingCoins = cva.GetVestingCoins(now.Add(18 * time.Hour))
+	require.Equal(t, sdk.Coins{sdk.NewInt64Coin(feeDenom, 250), sdk.NewInt64Coin(stakeDenom, 25)}, vestingCoins)
+}
+
+func TestSpendableCoinsCliffVestingAcc(t *testing.T) {
+	now := tmtime.Now()
+	cliffTime := now.Add(12 * time.Hour)
+	endTime := now.Add(24 * time.Hour)
+
+	bacc, origCoins := initBaseAccount()
+	cva := types.NewCliffVestingAccount(bacc, origCoins, now.Unix(), cliffTime.Unix(), endTime.Unix())
+
+	// require that all original coins are locked at the end of the vesting
+	// schedule
+	lockedCoins := cva.LockedCoins(now)
+	require.Equal(t, origCoins, lockedCoins)
+
+	// require that all original coins are locked at the end of the cliff vesting
+	// schedule
+	lockedCoins = cva.LockedCoins(cliffTime)
+	require.Equal(t, origCoins, lockedCoins)
+
+	// require that there exist no locked coins in the beginning of the
```

Comment:
I think the sentence is missing an end 😄

Reply:
Ah ah, copy paste from cosmos-sdk with the same ^^ https://github.com/cosmos/cosmos-sdk/blob/v0.46.1/x/auth/vesting/types/vesting_account_test.go#L95
It's fixed on my side now 😜.